app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +25,16 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
-      ...state,
-      users: action.data.course.users,
-      isLoaded: true,
-      lastRequestTimestamp: Date.now()
-    };
+    case RECEIVE_USERS: {
+     const sortedUsers = sortByKey(action.data.course.users, state.sort.initialSortKey, 'username' || state.sort.sortKey, null, false, false, false);
```

Comment:
Why are there so many arguments for `sortByKey` here, when there are fewer in the SORT_USERS action below?

Reply:
It looks like this still sorts by initialSortKey every time?

Reply:
It might be helpful to document the arguments or introduce constants to name them.

In this case, it looks like sortByKey already has logic for handling a `refresh` sort, where the sorting need not consider the previous key.

Reply:
Oh for the RECEIVE_USERS I added the default sort direction which I guess are not that necessary. I think I can do away with them and remain with only what's making a difference. 
I think it sorts by initialSortKey every time a specific key is not selected.